src/lua/uri.lua

```diff
@@ -49,6 +64,20 @@ local function parse(str)
             result[k] = ffi.string(uribuf[k])
         end
     end
+    result.options = {}
+    result.option_count = uribuf.option_count
+    for opt_idx = 0, result.option_count - 1 do
+        result.options[opt_idx + 1] = {}
+        result.options[opt_idx + 1].name =
+            ffi.string(uribuf.options[opt_idx].name)
+        result.options[opt_idx + 1].values = {}
+        result.options[opt_idx + 1].value_count =
+            uribuf.options[opt_idx].value_count
```

Comment:
You don't need to store `value_count` in a Lua table, because one can use `#values` to retrieve it.

Also, I think that the format should be different for Lua - parameters should be stored in a map, not in an array:

```lua
{
    ['param1'] = {'value1', 'value2', 'value3'},
    ['param2'] = {'value1'},
}
```

Reply:
How `&v` (without value) will be shown in Lua? As an empty table?

Reply:
```lua
uri.parse("/tmp/unix.sock?transport")

- host: unix/
  params:
    transport:
      values: []
  service: /tmp/unix.sock
  unix: /tmp/unix.sock
  query: transport
```

Reply:
Why do you need the `values` entry? I'd expect this:

uri.parse("/tmp/unix.sock?transport")

```lua
- host: unix/
  params:
    transport: []
  service: /tmp/unix.sock
  unix: /tmp/unix.sock
  query: transport
```